docs/cuda-target.md

```diff
@@ -97,7 +97,26 @@ The UniformState and UniformEntryPointParams struct typically vary by shader. Un
 
 ## Unsized arrays
 
-WIP: Not implemented yet.
+Unsized arrays can be used, which are indicated by an array with no size as in `[]`. For example 
+
+```
+    RWStructuredBuffer<int> arrayOfArrays[];
+```
+
+With normal 'sized' arrays, the elements are just stored contiguously within wherever they are defined. With an unsized array they map to `Array<T>` which is...
+
+```
+    T* data;
+    size_t count;
+```    
+
+Note that there is no method in the shader source to get the `count`, even though on the CUDA target it is stored and easily available. This is because of the behavior on GPU targets 
+
+* That the count has to be stored elsewhere (unlike with CUDA) 
+* On some GPU targets there is no bounds checking - accessing outside the bound values can cause *undefined behavior*
+* The elements may be laid out *contiguously* on GPU
```

Comment:
I'm not sure what a better term is here, but it seems clear that the elements are laid out contiguously above (all the elements are stored in contiguous memory pointed to by `data`). There's an additional detail of how the storage is "flat" or "tail-allocated" or "internal to the parent," but not none of those terms is really good at capturing the intention.

Also: eventually it would be good to clean up our behavior for unsized arrays to make sure they are consistent across targets.